### PR TITLE
docs(components): refactor examples with js code and framework icon c…

### LINKED ISSUE #515: Docs: add examples for ES6 javascript
currently all examples are written in typescript, it would be great to see examples of implementations in javascript

### PR DESCRIPTION
**Pull request by artyorsh**
…omponent

### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/react-native-ui-kitten/blob/master/CONTRIBUTING.md) guide.
 - [x] I read and followed the [New Feature Checklist](https://github.com/akveo/react-native-ui-kitten/blob/master/DEV_DOCS.md) guide.

 #### Short description of what this resolves:

Improves documentation with:
- Refactoring examples and markdown docs with JS code
- Refactoring examples with framework icon component (where possible to use)

Closes #515

### CODE REVIEW

File: src/framework/ui/button/button.component.tsx [hunk 1]
```diff
@@ -74,87 +74,69 @@ export type ButtonElement = React.ReactElement<ButtonProps>;
  *
  * ```
  * import React from 'react';
- * import {
- *   Button,
- *   ButtonProps,
- * } from 'react-native-ui-kitten';
+ * import { Button } from 'react-native-ui-kitten';
  *
- * export const ButtonShowcase = (props?: ButtonProps): React.ReactElement<ButtonProps> => {
```
Comment (32penkin) [inline src/framework/ui/button/button.component.tsx]: Basic icon example

**Comment by 32penkin** [comment]
Suggest adding examples without using a third-party module.

File: src/framework/ui/bottomNavigation/bottomNavigation.component.tsx [hunk 2]
```diff
@@ -61,57 +61,100 @@ export type BottomNavigationElement = React.ReactElement<BottomNavigationProps>;
  * @overview-example Simple Usage
  *
  * ```
+ * // IMPORTANT: To use Icon component make sure to follow this guide:
```
[the remark this answers is not in the document]
Comment (artyorsh) [reply src/framework/ui/bottomNavigation/bottomNavigation.component.tsx]: Good point. Will do

File: src/framework/ui/button/button.component.tsx (hunk 1 above)
Comment (artyorsh) [reply src/framework/ui/button/button.component.tsx]: There is `With Icon` example

File: src/framework/ui/icon/icon.component.tsx [hunk 3]
```diff
@@ -27,6 +27,7 @@ export type IconElement<T> = React.ReactElement<T>;
  * `Icon` component with animation support. Allows to render any ReactElement registered for a specific name.
```
Comment (32penkin) [inline src/framework/ui/icon/icon.component.tsx]: I believe it will be better to add 'asset-using' examples to all components where it can be.

**Pull request merged**